demos/usciter/res/live-reload.tis

```diff
@@ -0,0 +1,127 @@
+
+class LiveReload
+{
+  function this(on_reload) {
+    this.enabled = false;
+    this.urls = {};
+    this.watchers = {};
+    this.onRequestResponse = null;
+    this.onReload = on_reload;
+    this.delayReload = function() {
+      return (this super).reload();
+    };
+  }
+
+  // PUBLIC:
+
+  function stop() {
+    this.enabled = false;
+    self.timer(0, this.delayReload);
+  }
+
+  function start() {
+    this.enabled = true;
+    this.monitorRequests();
+  }
+
+  function reset() {
+    // clear already tracked URLs
+    // and wait for the new ones.
+    this.urls = {};
+  }
+
+  // PRIVATE:
+
+  function reload() {
+    if(!this.enabled)
+      return;
+
+    // reload the current file
+    debug(live: reload);
+    this.reset();
+    if(this.onReload)
+      this.onReload();
+  }
+
+  function scheduleReload() {
+    self.timer(3000, this.delayReload);
+  }
+
+  function addFile(url) {
+    if(this.urls[url])
+      return;
+
+    var path = URL.toPath(url /~ "/").toLowerCase();
+
+    this.urls[url] = path;
+    debug(live: add {url} as {path});
+
+    if(!this.watchers[path]) {
+      this.watchers[path] =
+
+        System.watch(path) << event change (filename) {
+          var path = this.path + "/" + filename;
+          debug(live: change in {path});
```

Comment:
@c-smile Andrew, is there any reason that `debug` doesn't work in this place? I've tried to add different debug statements, neither of them work. That's why `stdout` is there.

Reply:
Not sure, the only thing: debug message queue is associated with debugee document. When you reload document the queue is dropped.

Reply:
No, if I comment the reloading, it still doesn't print debug messages..

Reply:
The live-reload feature landed in r7741 + associated fixes.

Reply:
I've checked it. Looks great. Thanks!